### PR TITLE
ref(tests): Convert Sentry App Details Modal to rtl + typescript

### PR DESCRIPTION
**Pull request by priscilawebdev**
Convert test to rtl + typescript

### CODE REVIEW

File: static/app/components/modals/sentryAppDetailsModal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        onInstall={jest.fn()}
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(screen.getByRole('button', {name: 'Accept & Install'})).toBeDisabled();
+  });
+
+  it('when the Integration requires no permissions, does not render permissions', function () {
```
Comment (evanpurkhiser) [inline static/app/components/modals/sentryAppDetailsModal.spec.tsx:155]: I think the sentence needs reversed here so the “it” prefix works.

right now it’s “it when …”

File: static/app/components/modals/sentryAppDetailsModal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        sentryApp={sentryApp}
+      />
+    );
+
+    userEvent.click(screen.getByRole('button', {name: 'Accept & Install'}));
+
+    expect(handleOnInstall).toHaveBeenCalled();
+  });
+
+  it('when the User does not have permission to install Integrations, does not display the Install button', function () {
```
Comment (priscilawebdev) [inline static/app/components/modals/sentryAppDetailsModal.spec.tsx:121]:
```suggestion
  it('does not display the Install button, when the User does not have permission to install Integrations', function () {
```

File: static/app/components/modals/sentryAppDetailsModal.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(
+      screen.queryByRole('button', {name: 'Accept & Install'})
+    ).not.toBeInTheDocument();
+  });
+
+  it('when the Integration is installed, render the Install button disabled', function () {
```
Comment (priscilawebdev) [inline static/app/components/modals/sentryAppDetailsModal.spec.tsx:139]:
```suggestion
  it('render the Install button disabled, when the Integration is installed', function () {
```

File: static/app/components/modals/sentryAppDetailsModal.spec.tsx (hunk 1 above)
Comment (priscilawebdev) [inline static/app/components/modals/sentryAppDetailsModal.spec.tsx:155]:
```suggestion
  it('does not render permissions, when the Integration requires no permissions', function () {
```

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**